specs/Extensions.md

```diff
@@ -0,0 +1,585 @@
+    openFileName.hwndOwner = nullptr;
+    openFileName.hInstance = nullptr;
+    WCHAR fileName[MAX_PATH] = L"";
+    openFileName.lpstrFile = fileName;
+    openFileName.lpstrFilter = L"Manifest\0manifest.json\0\0";
+    openFileName.nMaxFile = ARRAYSIZE(fileName);
+    openFileName.Flags = OFN_OVERWRITEPROMPT;
+
+    if (GetOpenFileName(&openFileName))
+    {
+        // Remove the filename part of the path.
+        *wcsrchr(fileName, L'\\') = L'\0';
+        profile6->AddBrowserExtension(
+            fileName,
+            Callback<ICoreWebView2ProfileAddBrowserExtensionCompletedHandler>(
+                [](HRESULT error, ICoreWebViewBrowserExtension* extension) -> HRESULT
+                {
+                    if (error != S_OK)
+                    {
+                        ShowFailure(error, L"Failed to add extension");
+                    }
+                    wil::unique_cotaskmem_string id;
+                    extension->get_Id(&id);
+
+                    wil::unique_cotaskmem_string name;
+                    extension->get_Name(&name);
+
+                    std::wstring extensionInfo;
+                    extensionInfo.append(L"Added extension ");
+                    extensionInfo.append(name.get());
+                    extensionInfo.append(L" (");
+                    extensionInfo.append(id.get());
+                    extensionInfo.append(L")");
+                    MessageBox(nullptr, extensionInfo.c_str(), L"AddBrowserExtension Result", MB_OK);
+                    return S_OK;
+                })
+                .Get());
+    }
+}
+
+void BrowserExtensionService::RemoveExtension()
+{
+    // Get the profile object.
+    auto webView2_13 = m_webView.try_query<ICoreWebView2_13>();
+    wil::com_ptr<ICoreWebView2Profile> webView2Profile;
+    CHECK_FAILURE(webView2_13->get_Profile(&webView2Profile));
+    auto profile6 = webView2Profile.try_query<ICoreWebView2Profile6>();
+
+    profile6->GetExtensions(
+        Callback<ICoreWebView2ProfileGetBrowserExtensionCompletedHandler>(
+            [this](HRESULT error, ICoreWebView2BrowserExtensionList* extensions) -> HRESULT
+            {
+                TextInputDialog dialog(
+                    m_appWindow->GetMainWindow(), L"Remove Extension",
+                    L"Extension ID:", extensionIdString.c_str());
+                if (dialog.confirmed)
+                {
+                    for (UINT index = 0; index < extensionsCount; ++index)
+                    {
+                        wil::com_ptr<ICoreWebView2BrowserExtension> extension;
+                        extensions->GetValueAtIndex(index, &extension);
+
+                        wil::unique_cotaskmem_string id;
+                        wil::unique_cotaskmem_string name;
+
+                        extension->get_Id(&id);
+                        if (_wcsicmp(id.get(), dialog.input.c_str()) == 0)
+                        {
+                            extension->Remove(
```

Comment:
Presumably we are assuming that all WebViews sharing this profile are coordinating their changes to the extensions list, so that you don't have problems like double-remove throwing an exception.